### PR TITLE
Fix/logging scalp scenarios

### PR DESCRIPTION
**Pull request by katedmc**
# Description

What: Add logging in to SCALP to keep track of which scenarios the user has selected.
Why: This will allow us better tracking of the user's interaction with SCALP and navigating potential errors.
How: A logger and file handler have been added.

PR review time: 15-30 minutes
PR testing time: 15-30 minutes

## Added

- Added a logger
- Added a file handler
- Added a method to initialize the file handler with logger (Line 566 - 588)
- Added a constant to represent maximum log file size (Line 123)
- Added scenario logging call directly after user has chosen scenario (Line 462)

## Changed

-Nothing.

## Removed

-Nothing.

# How should this be manually tested?

1. Create tester class inside the enamel package with a main method
2. Populate main method with following code:
		args = new String[2];
		args[0] = "START_FACTORY";
		args[1] = \\\*INSERT ABSOLUTE PATH TO FACTORY SCENARIOS HERE\*\\; 
		SCALP.main(args);
3. Determine absolute path to factory scenarios
4. Replace commented block with absolute path.
5. Run tester and follow prompts.
6. Select any scenario
7. Check logs directory for "userScenarios.log.txt"
8. Confirm log file has been updated with recent scenario.
9. Remove tester class

### CODE REVIEW

**Review by saattalla — APPROVED** [review]

File: src/enamel/SCALP.java [hunk 1]
```diff
@@ -539,4 +563,27 @@ private void removeHWButtons() {
      void speak(String message) {
        voice.speak(message);
     }
+     
+     private void fileHanderInit() {
```
Comment (danielmapar) [inline src/enamel/SCALP.java]: The file handler function is good, but maybe we should move this to a separate class. I noticed other parts of the code that would make good use of this function.

Comment (danielmapar) [reply src/enamel/SCALP.java]: Maybe changing the function to receive a `this.logger` instance and a `Path` variable would make it more flexible and re-usable.

Comment (katedmc) [reply src/enamel/SCALP.java:567]: If there are no objections, I am going to discard this PR. This method is being moved to the logger class, and the two other relevant lines in the code will be included in the other logging PR after I commit the logging class, and then adjust the logging instances across the TBB.

Comment (katedmc) [reply src/enamel/SCALP.java:567]: Alternatively, I can update the other logging branch with the new Logger Class, merge it with the development branch, and then update this branch with its new intended code. I am happy with whichever option makes most sense from a workflow perspective.

**Pull request closed without merging**